week-2-wide-perspective/w02e02-data-manipulation/src/main.ts

```diff
@@ -0,0 +1,20 @@
+import { ordersFakeData } from "./orders/orders.fake-data";
+import { setAnnualOrders } from "./orders/tiles/AnnualOrdersTile";
+import { setBestSale } from "./orders/tiles/BestSaleTile";
+import { setTotalIncome } from "./orders/tiles/TotalIncomeTile";
+import { initAddOrderButton } from "./orders/tiles/AddOrderButton";
+import { Orders } from "./model/orders";
+import { setTotalOrders } from "./orders/tiles/TotalOrdersTile";
+
+const ordersStore = new Orders(ordersFakeData);
+
+const init = async () => {
+  const ordersList = await ordersStore.getAllOrders();
+  setAnnualOrders(ordersStore.getCurrentYear());
+  setBestSale(ordersStore.getBestSale());
+  setTotalIncome(ordersStore.getTotalIncome());
+  setTotalOrders(ordersList);
+  initAddOrderButton(ordersStore);
+};
```

Comment:
To rozwiązanie najmniej mi się podoba z całego kodu (reszta koncepcyjnie jest bardzo w porządku, fajowo że korzystasz z `RxJS` i wiesz jak sobie `Ramde` wpiąć we własne funkcje i zastosować kompozycję.

Tutaj `init` robi za dużo.
Powinien zainicjować każdy komponent ale dać mu pełny `ordersStore`
i wtedy każdy komponent SAM zdecyduje do jakiego strumienia ordersStore się wpina i subskrybuje zmiany.

Tutaj `init` podejmuje niepotrzebnie za dużo decycji o tym kto co dostanie.
Komponent sam sobie to powinien "ogarniać"